chains/ethereum/connection.go

```diff
@@ -24,6 +25,13 @@ import (
 
 var _ chains.Connection = &Connection{}
 
+var NonceLock = &sync.Mutex{}
+
+type Nonce struct {
+	nonce uint64
+	lock  *sync.Mutex
+}
+
```

Comment:
Can we add some comments about the race condition so future users understand why this is implemented?